### PR TITLE
refactor cuistl to gpuistl

### PR DESCRIPTION
**Pull request by multitalentloes**
Refactoring cuistl to gpuistl. This emphasizes that we are not solely supporting Nvidia GPUs anymore.

The cuistl classes under ::detail will be refactored later if necessary. These are mainly classes to cooperate closely with cuda libraries, so renaming those might obfuscate the code for now.

specific files in details/ should also refactored, in particular files like cusparse_matrix_operations now that cusparsematrix is refactored.

### CODE REVIEW

**Comment by multitalentloes** [comment]
Jenkins build this plase

**Comment by akva2** [comment]
jenkins build this please

**Comment by multitalentloes** [comment]
jenkins build this please

**Review by kjetilly — COMMENTED** [review]
Minor typos

File: opm/simulators/linalg/PreconditionerFactory_impl.hpp [hunk 1]
```diff
@@ -582,68 +582,68 @@ struct StandardPreconditioners<Operator, Dune::Amg::SequentialInformation> {
...
             const double w = prm.get<double>("relaxation", 1.0);
             using block_type = typename V::block_type;
             using VTo = Dune::BlockVector<Dune::FieldVector<float, block_type::dimension>>;
             using matrix_type_to =
                 typename Dune::BCRSMatrix<Dune::FieldMatrix<float, block_type::dimension, block_type::dimension>>;
-            using CuILU0 = typename cuistl::
-                CuSeqILU0<matrix_type_to, cuistl::CuVector<float>, cuistl::CuVector<float>>;
-            using Adapter = typename cuistl::PreconditionerAdapter<VTo, VTo, CuILU0>;
-            using Converter = typename cuistl::PreconditionerConvertFieldTypeAdapter<Adapter, M, V, V>;
+            using GpuuILU0 = typename gpuistl::
```
[the remark this answers is not in the document]
Comment (multitalentloes) [reply opm/simulators/linalg/PreconditionerFactory_impl.hpp]: fixed

**Comment by multitalentloes** [comment]
jenkins build this plase

**Comment by multitalentloes** [comment]
jenkins build this please

**Review by kjetilly — APPROVED** [review]
Waiting for Jenkins to complete.

**Pull request merged**